### PR TITLE
Adding Chat Tokens

### PR DESCRIPTION
**Pull request by clone1018**
Input Message: `Hello https://glimesh.tv :glimwow: world! How:glimlove:are https://google.com you!` 

Config: allow_links = true, allow_emotes = true

Output:
```json
{
    "message": "Hello https://glimesh.tv :glimwow: world! How:glimlove:are https://google.com you!",
    "tokens": [
        {
            "text": "Hello ",
            "type": "text"
        },
        {
            "text": "https://glimesh.tv",
            "type": "url",
            "url": "https://glimesh.tv"
        },
        {
            "text": " ",
            "type": "text"
        },
        {
            "src": "/emotes/svg/glimwow.svg",
            "text": ":glimwow:",
            "type": "emote",
            "url": "https://glimesh.dev/emotes/svg/glimwow.svg"
        },
        {
            "text": " world! How",
            "type": "text"
        },
        {
            "src": "/emotes/svg/glimlove.svg",
            "text": ":glimlove:",
            "type": "emote",
            "url": "https://glimesh.dev/emotes/svg/glimlove.svg"
        },
        {
            "text": "are ",
            "type": "text"
        },
        {
            "text": "https://google.com",
            "type": "url",
            "url": "https://google.com"
        },
        {
            "text": " you!",
            "type": "text"
        }
    ]
}
```

### CODE REVIEW

**Comment by clone1018** [comment]
I suspect I need to make animated emotes work!

File: lib/glimesh_web/live/chat_live/index.html.leex [hunk 1]
```diff
@@ -1,45 +1,35 @@
...
-                    datetime="<%= "#{chat_message.inserted_at}" <> "Z" %>" format="micro" hour="numeric"
-                    minute="2-digit" second="2-digit"><%= NaiveDateTime.to_time(chat_message.inserted_at) %>
+                <local-time id="timestamp-<%= chat_message.id %>" phx-update="ignore" datetime="<%= "#{chat_message.inserted_at}" <> "Z" %>" format="micro" hour="numeric" minute="2-digit" second="2-digit"><%= NaiveDateTime.to_time(chat_message.inserted_at) %>
                 </local-time>
             </small>
         </div>
         <div class="user-message">
             <%= if chat_message.is_followed_message, do: Glimesh.Chat.Effects.render_username_and_avatar(chat_message.user) %>
-            <%= Glimesh.Chat.Parser.parse_and_render(chat_message, @channel_chat_parser_config) %>
+            <%= raw(Glimesh.Chat.Renderer.render(chat_message.tokens)) %>
```
Comment (clone1018) [inline lib/glimesh_web/live/chat_live/index.html.leex:32]: Considering if this needs to be a raw or not...

Comment (mja00) [reply lib/glimesh_web/live/chat_live/index.html.leex:32]: Needs to be raw, otherwise, it makes links show the raw a tag

Comment (clone1018) [reply lib/glimesh_web/live/chat_live/index.html.leex:32]: Ah yeah, because I pre-convert it into safe in the renderer!

**Review by mja00 — APPROVED** [review]

**Pull request merged**